internal/configs/provider.go

```diff
@@ -70,17 +77,25 @@ func decodeProviderBlock(block *hcl.Block) (*Provider, hcl.Diagnostics) {
 	}
 
 	if attr, exists := content.Attributes["alias"]; exists {
-		valDiags := gohcl.DecodeExpression(attr.Expr, nil, &provider.Alias)
-		diags = append(diags, valDiags...)
+		provider.AliasExpr = attr.Expr
 		provider.AliasRange = attr.Expr.Range().Ptr()
+	}
 
-		if !hclsyntax.ValidIdentifier(provider.Alias) {
-			diags = append(diags, &hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  "Invalid provider configuration alias",
-				Detail:   fmt.Sprintf("An alias must be a valid name. %s", badIdentifierDetail),
-			})
-		}
+	if attr, exists := content.Attributes["for_each"]; exists {
+		provider.ForEach = attr.Expr
+	}
+
+	if attr, exists := content.Attributes["count"]; exists {
+		provider.Count = attr.Expr
+	}
+
+	if provider.Count != nil && provider.ForEach != nil && provider.AliasExpr != nil {
```

Comment:
This expression is incorrect

Reply:
yeah, i thought that when writing it. now to remember the logic lecture from first year of uni.

Reply:
I'd recommend something like:
```
if (alias != nil && count != nil) || (alias != nil && for_each != nil) || (count != nill && for_each != nil) {
  // common error message
}
```